packages/core/src/domain/session/sessionStoreOperations.ts

```diff
@@ -0,0 +1,109 @@
+export const LOCK_MAX_TRIES = 100
+const bufferedOperations: Operations[] = []
+let ongoingOperations: Operations | undefined
+
+export function processSessionStoreOperations(operations: Operations, sessionStore: SessionStore, numberOfRetries = 0) {
+  const { retrieveSession, persistSession, clearSession } = sessionStore
+  const lockEnabled = isLockEnabled()
+
+  if (!ongoingOperations) {
+    ongoingOperations = operations
+  }
+  if (operations !== ongoingOperations) {
+    bufferedOperations.push(operations)
+    return
+  }
+  if (lockEnabled && numberOfRetries >= LOCK_MAX_TRIES) {
+    next(sessionStore)
+    return
+  }
+  let currentLock: string
+  let currentSession = retrieveSession()
+  if (lockEnabled) {
+    // if someone has lock, retry later
+    if (currentSession.lock) {
+      retryLater(operations, sessionStore, numberOfRetries)
+      return
+    }
+    // acquire lock
+    currentLock = generateUUID()
+    currentSession.lock = currentLock
+    persistSession(currentSession)
+    // if lock is not acquired, retry later
+    currentSession = retrieveSession()
+    if (currentSession.lock !== currentLock) {
+      retryLater(operations, sessionStore, numberOfRetries)
+      return
+    }
+  }
+  let processedSession = operations.process(currentSession)
+  if (lockEnabled) {
+    // if lock corrupted after process, retry later
+    currentSession = retrieveSession()
+    if (currentSession.lock !== currentLock!) {
+      retryLater(operations, sessionStore, numberOfRetries)
+      return
+    }
+  }
+  if (processedSession) {
+    if (isSessionInExpiredState(processedSession)) {
+      clearSession()
+    } else {
+      processedSession.expire = String(dateNow() + SESSION_EXPIRATION_DELAY)
+      persistSession(processedSession)
+    }
+  }
+  if (lockEnabled) {
+    // correctly handle lock around expiration would require to handle this case properly at several levels
+    // since we don't have evidence of lock issues around expiration, let's just not do the corruption check for it
+    if (!(processedSession && isSessionInExpiredState(processedSession))) {
+      // if lock corrupted after persist, retry later
+      currentSession = retrieveSession()
+      if (currentSession.lock !== currentLock!) {
+        retryLater(operations, sessionStore, numberOfRetries)
+        return
+      }
+      delete currentSession.lock
+      persistSession(currentSession)
+      processedSession = currentSession
+    }
+  }
+  // call after even if session is not persisted in order to perform operations on
+  // up-to-date session state value => the value could have been modified by another tab
+  operations.after?.(processedSession || currentSession)
+  next(sessionStore)
+}
+/**
+ * Lock strategy allows mitigating issues due to concurrent access to cookie.
+ * This issue concerns only chromium browsers and enabling this on firefox increases cookie write failures.
+ */
+
+export const isLockEnabled = () => isChromium()
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏spacing 🙈
```suggestion
}

/**
 * Lock strategy allows mitigating issues due to concurrent access to cookie.
 * This issue concerns only chromium browsers and enabling this on firefox increases cookie write failures.
 */
export const isLockEnabled = () => isChromium()
```

Reply:
🙈